File: src/main/scala/com/exasol/cloudetl/bucket/LocalBucket.scala
package com.exasol.cloudetl.bucket

import org.apache.hadoop.conf.Configuration

/**
 * A specific [[Bucket]] implementation for the local 'file:' scheme.
 */
final case class LocalBucket(path: String, params: Map[String, String]) extends Bucket {

  /** @inheritdoc */
  override val bucketPath: String = path

  /** @inheritdoc */
  override val properties: Map[String, String] = params

  /** @inheritdoc */
  override def validate(): Unit = ()

  /** @inheritdoc */
  override def getConfiguration(): Configuration = {
    validate()
    new Configuration()
  }
}

File: src/main/java/com/exasol/cloudetl/bucket/LocalBucket.java
package com.exasol.cloudetl.bucket;

import java.util.List;

import org.apache.hadoop.conf.Configuration;

import com.exasol.cloudetl.ScalaConverters;
import com.exasol.cloudetl.storage.StorageProperties;

/** Bucket implementation for local file paths. */
public final class LocalBucket extends AbstractConfiguredBucket {
    /** Create a bucket. */
    public LocalBucket(final String path, final StorageProperties params) {
        super(path, params);
    }

    @Override
    public scala.collection.immutable.Seq<String> getRequiredProperties() {
        return ScalaConverters.seqFromJava(List.of());
    }

    @Override
    public void validate() {
        validateRequiredProperties();
    }

    @Override
    public Configuration getConfiguration() {
        validate();
        return new Configuration();
    }
}
